### PR TITLE
[Merged by Bors] - tortoise: refactor initialization and configuration for tortoise

### LINKED ISSUE #2707: Separate tortoise config
Right now, tortoise config lives inside the main app config, and is assembled by the App before the verifying tortoise is instantiated:

https://github.com/spacemeshos/go-spacemesh/blob/2d87075870e3057958517f3b3544fa6138b9ea19/cmd/node/node.go#L624-L638

The tortoise should have its own config module. `TortoiseRerunInterval` should be switched from an int to time.Duration so that it can be specified as, e.g., "12h" in the config.

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

Config for a tortoise is a bit messy, it includes both mandatory deps (like mesh db, beacons, atx db) and immutable protocol parameters as a part of the same struct. 

closes: https://github.com/spacemeshos/go-spacemesh/issues/2707

## Changes
- refactor tortoise module to have cleaner initialization (New function, and options for non-mandatory parameters)
- refactor tortoise config to avoid duplication and use it when initializing node

## Test Plan
ut

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: tortoise/algorithm.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		t.cfg,
+	)
+
+	if err := t.trtl.Recover(); err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			t.trtl.init(t.ctx, mesh.GenesisLayer())
+		} else {
+			t.logger.With().Panic("can't recover turtle state", log.Err(err))
+		}
+	}
```
Comment (nkryuchkov) [inline tortoise/algorithm.go]: What do you think about removing the `else` statement?
```suggestion
	if err := t.trtl.Recover(); err != nil {
		if !errors.Is(err, database.ErrNotFound) {
			t.logger.With().Panic("can't recover turtle state", log.Err(err))
		}
		t.trtl.init(t.ctx, mesh.GenesisLayer())
	}
```

**Review by nkryuchkov — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
looks like setup issue `Exception: genesis_delta time=75sec, is too short for this deployment. delta_from_genesis=-47.795942`

bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**